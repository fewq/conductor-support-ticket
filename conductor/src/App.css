.App {
  text-align: center;
}

.App-logo {
  animation: App-logo-spin infinite 100s linear;
  height: 40vmin;
  pointer-events: none;
}

.App-header {
  background-color: #282c34;
  min-height: 100vh;
  display: flex;
  flex-direction: column;
  align-items: center;
  justify-content: center;
  font-size: calc(10px + 2vmin);
  color: white;
}

.App-link {
  color: #61dafb;
}

@keyframes App-logo-spin {
  from {
    transform: rotate(0deg);
  }
  to {
    transform: rotate(360deg);
  }
}

.subtitle {
  color: #f6c645;
}

/* 
.App body {
  background-color: #282c34;
  color: white;
  text-align: center;
  font-family: -apple-system, BlinkMacSystemFont, "Segoe UI", "Roboto", "Oxygen",
    "Ubuntu", "Cantarell", "Fira Sans", "Droid Sans", "Helvetica Neue",
    sans-serif;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  padding: 0;
}
*/

.body {
  background-color: #282c34;
  color: white;
  text-align: center;
  font-family: -apple-system, BlinkMacSystemFont, "Segoe UI", "Roboto", "Oxygen",
    "Ubuntu", "Cantarell", "Fira Sans", "Droid Sans", "Helvetica Neue",
    sans-serif;
  -webkit-font-smoothing: antialiased;
  -moz-osx-font-smoothing: grayscale;
  padding: 0;
}

/* Thad defined buttons here */
/* 
input,
button {
  font-family: "Nunito", sans-serif;
  font-weight: 900;
}

.App input {
  display: block;
  border: 1px solid #ccc;
  border-radius: 5px;
  background: #fff;
  padding: 15px;
  outline: none;
  width: 20%;
  margin-bottom: 20px;
  transition: 0.3s;
  margin-left: auto;
  margin-right: auto;
}

.App button {
  background: #5d8ffc;
  color: #fff;
  border: 1px solid #5d8ffc;
  border-radius: 5px;
  padding: 15px;
  display: block;
  width: 20%;
  transition: 0.3s;
  margin-left: auto;
  margin-right: auto;
}

.App form {
  width: 100%;
  display: block;
} */
