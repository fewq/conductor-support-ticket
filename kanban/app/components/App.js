import React,{PropTypes} from "react";
import {connect} from "react-redux";
import Board from "./Board";
import AddList from "./AddList";
import ListContainer from "../containers/ListContainer";
import {closeAllPopups, showEditor, createList} from "../actions/";

const mapStateToProps = ({domainData, appState}) => ({
	allLists : domainData.lists.allLists,
	itemToEdit : appState.itemToEdit,
	attributeToEdit : appState.attributeToEdit,
});

const mapDispatchToProps = (dispatch) => ({
	onClickAddAList : (e) => {e.stopPropagation(); dispatch(showEditor("board", "new list"));},
	onBodyClick : () => dispatch( closeAllPopups() ),
	onClickSaveList : (listName) => dispatch(createList(listName))
});

const AppPropTypes = {
	allLists : PropTypes.array.isRequired,
	onBodyClick : PropTypes.func.isRequired,
	onClickAddAList : PropTypes.func.isRequired,
	onClickSaveList : PropTypes.func.isRequired,
	itemToEdit : PropTypes.oneOfType([PropTypes.string, PropTypes.number ]).isRequired,
	attributeToEdit : PropTypes.string.isRequired
};

const App = (props) => (
	<div className="root-comp" onClick={props.onBodyClick}>
		<header>
			<a href="https://github.com/Dajust/reduxTrello" target="blank" className="source-code">Browse the code</a>
			<h1 className="app-name">reduxTrello</h1>
		</header>
		<Board>
			{
				props.allLists.map((id, i) =>
					<ListContainer
						id = {id}
						key= {id}
						index={i}
						/>
				)
			}
			<AddList
				onClickAddAList = {props.onClickAddAList}
				onClickSaveList = {props.onClickSaveList}
				itemToEdit = {props.itemToEdit}
				attributeToEdit={props.attributeToEdit}
				/>
		</Board>
	</div>
);

App.propTypes = AppPropTypes;

export default connect(mapStateToProps, mapDispatchToProps)(App);
