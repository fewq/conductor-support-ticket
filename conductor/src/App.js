import React, { Component } from "react";
import logo from "./logo.svg";
import "./App.css";
import Main from "./components/Main";
import Restricted from "./components/Restricted";
import NotFound from "./components/NotFound";
import Callback from "./components/Callback";

class App extends Component {
  render() {
    // mainComponent's default value is an empty string
    let mainComponent = "";
    // This switch statement decides what to load in the body of the app
    switch (this.props.location) {
      case "":
        // this case is the homepage
        console.log("case '': Loading <Main /> or homepage ");
        mainComponent = <Main {...this.props} />;
        break;
      case "callback":
        //this case is when the app is in the process of authenication
        console.log("case callback: Loading the loading page");
        mainComponent = <Callback />;
        break;
      case "restricted":
        // this case can only be accessed if the user is authenticated
        // to prevent misuse, it uses the isAuthenticated method in Auth.js
        // if not authenticated, it loads the <NotFound /> page instead
        console.log("case restricted: loading <Restricted /> or <NotFound />");
        mainComponent = this.props.auth.isAuthenticated() ? (
          <Restricted {...this.props} />
        ) : (
          <NotFound />
        );
        break;
      default:
        console.log("case default: loading <NotFound />");
        mainComponent = <NotFound />;
    }

    return (
      <div className="App">
        <header className="App-header">
          <img src={logo} className="App-logo" alt="logo" />
          Welcome to the <b>ACNAPI Modern Ticket Support System</b>
          <p class="subtitle">Powered by the Conductor</p>
        </header>
        <body>
          {/* This is where the actual components are loaded */}
          {mainComponent}
        </body>
      </div>
    );
  }
}

export default App;
