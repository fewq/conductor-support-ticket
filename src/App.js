import React, { Component } from "react";
import logo from "./logo.svg";
import "./App.css";
import Main from "./components/Main";
import Restricted from "./components/Restricted";
import NotFound from "./components/NotFound";
import Callback from "./components/Callback";
require("dotenv").config();

class App extends Component {
    render() {
        // mainComponent's default value is an empty string
        let mainComponent = "";
        // This switch statement decides what to load in the body of the app
        switch (this.props.location) {
            case "":
                // this case is the homepage
                console.log("case '': Loading <Main /> or homepage ");
                mainComponent = <Main {...this.props} />;
                break;
            case "callback":
                //this case is when the app is in the process of authenication
                console.log("case callback: Loading the loading page");
                mainComponent = <Callback />;
                break;
            case "restricted":
                // this case can only be accessed if the user is authenticated
                // to prevent misuse, it uses the isAuthenticated method in Auth.js
                // if not authenticated, it loads the <NotFound /> page instead
                console.log(
                    "case restricted: loading <Restricted /> or <NotFound />"
                );
                mainComponent = this.props.auth.isAuthenticated() ? (
                    <Restricted {...this.props} />
                ) : (
                    <NotFound />
                );
                break;
            default:
                console.log("case default: redirecting to login");
                mainComponent = <Main {...this.props} />;
        }

        return (
            <main className="App">
                <header className="App-header">
                    <img src={logo} className="App-logo" alt="logo" />
                    <b>ACNAPI Ticket Support System</b>
                    <p className="subtitle">Powered by the Conductor project</p>
                </header>
                <div className="App-body">
                    {/* This is where the actual components are loaded */}
                    {mainComponent}
                </div>
            </main>
        );
    }
}

export default App;
