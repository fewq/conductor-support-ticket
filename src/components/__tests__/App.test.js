import React from 'react';
import ReactDOM from 'react-dom';
import App from '../../App';
import { render, fireEvent, cleanup, queryByTestId, getByDisplayValue, getByValue } from 'react-testing-library'
import {toBeInTheDocument, toHaveClass} from 'jest-dom'
expect.extend({toBeInTheDocument, toHaveClass})

afterEach(cleanup)

let mockAuth = {
  isAuthenticated: () => true
};

// tests
describe("Basic UI tests", () => {
  test('Renders without crashing', () => {
    const div = document.createElement('div');
    ReactDOM.render(<App auth={mockAuth}  />, div);
    ReactDOM.unmountComponentAtNode(div);
  });
  
  test('Renders app header: title, subtitle, logo', () => {
    const { getByText, getByAltText } = render(<App auth={mockAuth} />);
    let expectedText = {
      title: /ACNAPI Ticket Support System/i,
      subtitle: /Powered by the Conductor Project/i,
    }
    let appTitle = getByText(expectedText.title);
    let appSubtitle = getByText(expectedText.subtitle);
    let appLogo = getByAltText(/logo/);
    expect(appTitle).toBeDefined();
    expect(appSubtitle).toBeDefined();
    expect(appLogo).toBeDefined();
  })

  test('Styling of app header elements', () => {
    const { getByText, getByAltText } = render(<App auth={mockAuth} />);
    let expectedText = {
      title: /ACNAPI Ticket Support System/i,
      subtitle: /Powered by the Conductor Project/i,
    }
    let appSubtitle = getByText(expectedText.subtitle);
    let appLogo = getByAltText(/logo/);
    expect(appSubtitle).toHaveClass("subtitle");
    expect(appLogo).toHaveClass("App-logo");
  })
})

describe("Rendering App's first page under different conditions", () => {
  test('Default case: Main Component', () => {
    let location = "";
    const { getByText } = render(<App location={location} auth={mockAuth} />);
    const greetingText = getByText(/Hey,/i);
    expect(greetingText).toBeInTheDocument();
    
  });

  test('Authenticating: Callback Component', () => {
    let location = "callback";
    const { getByText } = render(<App location={location} auth={mockAuth} />);
    const loadingText = getByText(/Loading.../i);
    expect(loadingText).toBeInTheDocument();
  });

  test('Authenticated: Restricted Component', () => {
    let location = "restricted";
    const { getByText } = render(<App location={location} auth={mockAuth} />);
    const logOutButton = getByText(/Logout/i)
    expect(logOutButton).toBeInTheDocument();
  })

  test('Not Authenticated: NotFound Component', () => {
    let location = "restricted";
    let mockAuth = {
      isAuthenticated: () => false
    };
    const { getByText } = render(<App location={location} auth={mockAuth} />);
    let expectedText = getByText(/not found/i)
    expect(expectedText).toBeDefined();
  })

})

