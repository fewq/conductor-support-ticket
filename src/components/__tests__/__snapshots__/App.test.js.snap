// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`snapshot testing 1`] = `
<main
  className="App"
>
  <header
    className="App-header"
  >
    <img
      alt="logo"
      className="App-logo"
      src="logo.svg"
    />
    <b>
      ACNAPI Ticket Support System
    </b>
    <p
      className="subtitle"
    >
      Powered by the Conductor project
    </p>
  </header>
  <div
    className="App-body"
  >
    <div>
      <div>
        <p>
          Hey, 
          , how can we assist you today?
        </p>
        <p>
          Click 
          <a
            href="/restricted"
          >
             here
          </a>
           to access restricted area
        </p>
      </div>
    </div>
  </div>
</main>
`;
